pyvista/core/filters/data_set.py

```diff
@@ -48,9 +58,14 @@ def _clip_with_function(
         if return_clipped:
             a = _get_output(alg, oport=0)
             b = _get_output(alg, oport=1)
+            if crinkle:
+                a = self.extract_cells(np.unique(a.cell_data['cell_ids']))
+                b = self.extract_cells(np.unique(b.cell_data['cell_ids']))
```

Comment:
I am surprised that the DataSet class does not have a cell_id property.
Would it be difficult to add it [like the `ExplicitStructuredGrid` class](https://tkoyama010.github.io/pyvista-docs-dev-ja/api/core/_autosummary/pyvista.ExplicitStructuredGrid.cell_id.html#pyvista.ExplicitStructuredGrid.cell_id)?

Reply:
I would highly prefer not to be adding data to datasets unless absolutely necessary.  I work with a lot of very large datasets and often have to selectively read in data to get it to fit into memory.

It may be worth documenting when this is done under the hood as it will be an unexpected surprise to the user.

Reply:
Resolved in e4c9a2ac2b86ffc57822a06df9a3251eb69c639f